### PR TITLE
feat: fix parent issue

### LINKED ISSUE #268: Disable Bounties for "Parent Issues"
### Context

It's fast and easy to write todo lists in the original comment of a GitHub issue, and then immediately convert them to "child issues" directly from the GitHub user interface.

The problem I am seeing is that the bounty system breaks down where contributors can "double dip" on rewards easily by claiming bounties for the actual bounties (child issues) and then the "meta" or "parent" issue.

The argument possibly could be made for project management. However in the future all of the **issue creators**, and **commenters** (both are how the project manager would "manage") are to receive compensation for their contributions upon the completion of the bounty.

### Proposal

- When changing a label, the bot should check if there are linked issues in the original comment and disable setting a price (and most importantly, as a consequence, payment permit generation.)

- The bot should comment explaining why it refuses to set a price.

---

              Parent issues should not have bounties. Instead, the child issues should have bounties.

_Originally posted by @pavlovcik in https://github.com/ubiquity/grants/issues/5#issuecomment-1519067539_

### PR DESCRIPTION
**Pull request by wannacfuture**
Resolves #268 
QA: https://github.com/wannacfuture/Battleship/issues/65#issuecomment-1663270599

### CODE REVIEW

File: src/handlers/pricing/action.ts [hunk 1]
```diff
@@ -44,3 +50,8 @@ export const pricingLabelLogic = async (): Promise<void> => {
     logger.info(`Skipping action...`);
   }
 };
+
+export const isParentIssue = (body: string) => {
+  const parentPattern = /-\s+\[( |x)\]\s+#\d+/;
+  return body.match(parentPattern);
+};
```
Comment (0xcodercrane) [inline src/handlers/pricing/action.ts:57]: The regex seems limited to the pattern `- [ ] Subtask1`. once it is converted into an issue, it changes like `- https://github.com/ubiquity/ubquibot/issues/1`.

would you take a look again at my concern?

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]

File: src/handlers/pricing/action.ts (hunk 1 above)
Comment (wannacfuture) [reply src/handlers/pricing/action.ts:57]: Well, actually it it isn't. I just points out the issue number like #issue_number
image
image

**Comment by pavlovcik** [comment]
I'm not sure why GitHub merged this in but is this good to go? Should I close the linked issue?

**Comment by pavlovcik** [comment]
I think it's worth getting a QA test for this pull request before keeping the merged code!

**Comment by pavlovcik** [comment]
If anybody could help with testing that would be great 🙏

**Comment by wannacfuture** [comment]
I think @web4er  is working on QA for this pr.

**Comment by web4er** [comment]
QA: https://github.com/web4erOrg/ubiquibot/issues/17
Mostly working well. [1 test failed](https://github.com/web4erOrg/ubiquibot/issues/17#issuecomment-1672356785) as it didn't cover one proposal. And there is [one little concern](https://github.com/web4erOrg/ubiquibot/issues/17#issuecomment-1672344215). 
@pavlovcik

**Comment by pavlovcik** [comment]
Thanks so much for the excellent QA @web4er

**Comment by wannacfuture** [comment]
Whoa, Thanks! @web4er 
let me fix it.

**Pull request merged**